pipeline/notifications/check_message_test.py

```diff
@@ -34,23 +44,26 @@ def test_more_users_in_match():
     b_msg = Message(from_user="B", timestamp=datetime(2022, 7, 17))
     c_msg = Message(from_user="C", timestamp=datetime(2022, 7, 18))
     test_participants = {
-        "A": User(uid="A", displayName="Ayman"),
-        "B": User(uid="B", displayName="Brian"),
-        "C": User(uid="C", displayName="Chris"),
-        "D": User(uid="D", displayName="Dinora"),
-        "E": User(uid="E", displayName="Erik"),
+        "A": User(uid="A", displayName="Ayman", email="ayman@iit.edu"),
+        "B": User(uid="B", displayName="Brian", email="brian@iit.edu"),
+        "C": User(uid="C", displayName="Chris", email="chris@iit.edu"),
+        "D": User(uid="D", displayName="Dinora", email="dinora@iit.edu"),
+        "E": User(uid="E", displayName="Erik", email="erik@iit.edu"),
     }
     test_chatdata = ChatData(
         release_timestamp=datetime.now(),
         participants=test_participants,
         messages=[b_msg, c_msg],
     )
+    test_notif_inp = NotificationInput(
+        chatdata=test_chatdata, run_datetime=datetime.now()
+    )
     check_notifier = NewMessageNotifier()
-    actual = check_notifier.get_users_to_notify(test_chatdata)
+    actual = check_notifier.get_users_to_notify(test_notif_inp)
     assert actual == [
-        User(uid="A", displayName="Ayman"),
-        User(uid="D", displayName="Dinora"),
-        User(uid="E", displayName="Erik"),
+        NotificationInfo(name="Ayman", email="ayman@iit.edu"),
```

Comment:
I just realized that NotificationInfo doesn't have user ID, which we will need for simple filtering later on.

How about just having a `User` class nested in NotificationInfo?

```
@dataclass
class NotificationInfo:
  recipient: User
```

Could be called `recipient` or `for_user` or just `user`, though the last one might be too vague.

Reply:
Yeah I feel we can eliminate the "name" and "email" fields in favor of just having a "recipient" field that takes a User object. I still believe we should keep the "title" and "content" fields unless there's a reason those are superfluous as well.